eth/client.go

```diff
@@ -418,7 +419,174 @@ func (c *client) Bond(amount *big.Int, to ethcommon.Address) (*types.Transaction
 		}
 	}
 
-	return c.BondingManagerSession.Bond(amount, to)
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	// Get delegator
+	delegator, err := c.GetDelegator(sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	// Switching delegate's calculate old delegate positions
+	var oldHints lpTypes.TranscoderPoolHints
+	if delegator.DelegateAddress != to {
+		currentRound, err := c.CurrentRound()
+		if err != nil {
+			return nil, err
+		}
+
+		delegatorTotalStake, err := c.PendingStake(sender, currentRound)
+		if err != nil {
+			return nil, err
+		}
+		amount = new(big.Int).Add(delegatorTotalStake, amount)
+		// Get total bonded
+		totalBonded, err := c.TranscoderTotalStake(delegator.DelegateAddress)
+		if err != nil {
+			return nil, err
+		}
+		oldHints = simulateTranscoderPoolUpdate(delegator.DelegateAddress, new(big.Int).Sub(totalBonded, amount), transcoders, isFull)
```

Comment:
The old delegate's new stake should be `new(big.Int).Sub(totalBonded, delegatorTotalStake)`.

At the moment, `amount = delegatorTotalStake + amount` where `amount` is additional LPT being staked so `new(big.Int).Sub(totalBonded, amount)` results in subtracting too much from `totalBonded`.

Reply:
fixed in 13e5e37 and added a comment